# Chapter 18

In [1]:
import pandas as pd
import statsmodels.api as sm

In [2]:
# Exercise 1
wageprc = pd.read_stata("./stata/WAGEPRC.DTA")
X = sm.add_constant(wageprc[["gwage", "gprice_1"]])
model = sm.OLS(wageprc.gprice, X, missing="drop").fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 gprice   R-squared:                       0.454
Model:                            OLS   Adj. R-squared:                  0.450
Method:                 Least Squares   F-statistic:                     116.8
Date:                Thu, 27 Oct 2022   Prob (F-statistic):           1.21e-37
Time:                        23:47:34   Log-Likelihood:                 1298.5
No. Observations:                 284   AIC:                            -2591.
Df Residuals:                     281   BIC:                            -2580.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0013      0.000      4.675      0.000       0.001       0.002
gwage          0.0814      0.031      2.620      0.009       0.020       0.143
gprice_1       0.6396      0.045     14.331      0.000       0.552       0.727
==============================================================================
Omnibus:                       51.043   Durbin-Watson:                   2.363
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              378.052
Skew:                           0.433   Prob(JB):                     8.07e-83
Kurtosis:                       8.586   Cond. No.                         304.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [5]:
model.params[1] / (1 - model.params[2])

0.2259784476690603

In [6]:
X = sm.add_constant(wageprc[["gwage", "gprice_1", "gwage_1"]])
model = sm.OLS(wageprc.gprice, X, missing="drop").fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 gprice   R-squared:                       0.460
Model:                            OLS   Adj. R-squared:                  0.454
Method:                 Least Squares   F-statistic:                     79.44
Date:                Thu, 27 Oct 2022   Prob (F-statistic):           3.30e-37
Time:                        23:56:42   Log-Likelihood:                 1300.1
No. Observations:                 284   AIC:                            -2592.
Df Residuals:                     280   BIC:                            -2578.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0011      0.000      3.613      0.000       0.000       0.002
gwage          0.0898      0.031      2.866      0.004       0.028       0.151
gprice_1       0.6186      0.046     13.422      0.000       0.528       0.709
gwage_1        0.0551      0.032      1.740      0.083      -0.007       0.117
==============================================================================
Omnibus:                       50.774   Durbin-Watson:                   2.316
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              347.067
Skew:                           0.463   Prob(JB):                     4.32e-76
Kurtosis:                       8.336   Cond. No.                         325.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [8]:
(model.params[1] + model.params[3]) / (1 - model.params[2])

0.3799599608144597

C1.i Estimated IP is 0.081 and estimated LRP is 0.226.

C1.ii The IP in the previous problem is 0.119 and the LRP is 1.172. Both are much higher than the geometric distributed lag.

C1.iii The estimated IP is 0.09 and the LRP is 0.380. These are closer to (but high) (i) and still well below the results from 5.11 (previous problem).